encoder.go

```diff
@@ -390,8 +390,8 @@ func (encoder *encoder) addTruncation(reason TruncationReason, size int) {
 // mesureObjectDepth traverses the provided object recursively to try and obtain
 // the real object depth, but limits itself to about 1ms of time budget, past
 // which it'll stop and return whatever it has go to so far.
-func (encoder *encoder) measureObjectDepth(obj reflect.Value) {
-	ctx, cancelCtx := context.WithTimeout(context.Background(), time.Millisecond)
+func (encoder *encoder) measureObjectDepth(obj reflect.Value, timeout time.Duration) {
+	ctx, cancelCtx := context.WithTimeout(context.Background(), timeout)
```

Comment:
I'd suggest managing 0 as disabled for test purposes - or use infinite in the tests?

Reply:
I am not really a fan of arbitrary values and this does not seem necessary to complexify this code just for tests. I will just put a huge value in the tests

Reply:
I put `time.Hour` because there is no `time.Infinity`. The test and job timeout will trigger before this anyway